lib/utils.pm

```diff
@@ -1382,7 +1382,7 @@ sub get_root_console_tty {
     is running on tty2 by default. see also: bsc#1054782
 =cut
 sub get_x11_console_tty {
-    return (sle_version_at_least('15')) ? 2 : 7;
+    return ((is_sle && sle_version_at_least('15')) || (is_leap && leap_version_at_least('15'))) ? 2 : 7;
```

Comment:
the condition is wrong imho

It should be desktop = gnome, notauologin=1 and sle_version_at_least 15

Reply:
yes, that makes sense.

Reply:
trying with

```perl
(check_var('DESKTOP', 'gnome') && get_var('NOAUTOLOGIN') && \
    !(is_sle && !sle_version_at_least('15')  ) && \
    !(is_leap && !leap_version_at_least('15'))) ? 2 : 7
```